tests/validation/tests/v3_api/test_configmaps.py

```diff
@@ -0,0 +1,489 @@
+
+
+@if_test_rbac
+def test_rbac_cmap_project_member_create():
+    user_token = rbac_get_user_token_by_role(PROJECT_MEMBER)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, user_token)
+    rbac_configmap_create(p_client, ns)
+
+
+@if_test_rbac
+def test_rbac_cmap_project_member_edit():
+    user_token = rbac_get_user_token_by_role(PROJECT_MEMBER)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, user_token)
+    rbac_configmap_edit(p_client, ns)
+
+
+@if_test_rbac
+def test_rbac_cmap_project_member_delete():
+    user_token = rbac_get_user_token_by_role(PROJECT_MEMBER)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, user_token)
+    rbac_configmap_delete(p_client, ns)
+
+
+@if_test_rbac
+def test_rbac_cmap_project_readonly_member_create(remove_resource):
+
+    cluster_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+    project, ns = create_project_and_ns(cluster_owner_token,
+                                        namespace["cluster"],
+                                        random_test_name("rbac-test"))
+    user_token1 = rbac_get_user_token_by_role(PROJECT_READ_ONLY)
+    readonly_user_client = get_project_client_for_token(project, user_token1)
+
+    value = "valueall"
+    keyvaluepair = {"testall": value}
+
+    # Read Only member cannot create config maps
+    with pytest.raises(ApiError) as e:
+        create_configmap(keyvaluepair, readonly_user_client, ns)
+    assert e.value.error.status == 403
+    assert e.value.error.code == 'Forbidden'
+
+
+@if_test_rbac
+def test_rbac_cmap_project_readonly_member_edit(remove_resource):
+
+    cluster_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+    project, ns = create_project_and_ns(cluster_owner_token,
+                                        namespace["cluster"],
+                                        random_test_name("rbac-test"))
+    user_token1 = rbac_get_user_token_by_role(PROJECT_READ_ONLY)
+    readonly_user_client = get_project_client_for_token(project, user_token1)
+
+    value = "valueall"
+    keyvaluepair = {"testall": value}
+
+    cluster_owner_p_client = get_project_client_for_token(project,
+                                                          cluster_owner_token)
+    # As a cluster owner, create a config map
+    configmap = create_configmap(keyvaluepair, cluster_owner_p_client, ns)
+
+    # Readonly member cannot edit configmap
+    value1 = ("valueall")
+    value2 = ("valueallnew")
+    updated_dict = {"testall": value1, "testallnew": value2}
+
+    with pytest.raises(ApiError) as e:
+        readonly_user_client.update(configmap,
+                                    namespaceId=ns['name'],
+                                    data=updated_dict)
+    assert e.value.error.status == 403
+    assert e.value.error.code == 'Forbidden'
+
```

Comment:
need to delete the configmap at the end `remove_resource(configmap)`

Reply:
Done